### DISCUSSION TITLE
how to using regular expressions in safelist

### DISCUSSION THREAD

**Discussion by Shadecute** [tailwindlabs/tailwindcss#6585] [Help] [answered]
i'm  using white list in v2 And it worked well
`    options: {
            safelist: [/^bg-/, /^text-/, /^border-/]
}`

after upgrade to v3 This code no longer works,
i changed it:
`   safelist: [
        {
        pattern: [/^bg-/, /^text-/, /^border-/],
        variants: ['hover', 'focus'],
        },
        ]`
But it still does not work
please guide me

**Comment by leojpod** [comment, marked as answer]
Maybe try to make 1 pattern per regexp? 
like 
```
safelist: [ { pattern: [/^bg-/, variants: ... }, {pattern: /^text-/, ...}, {pattern: /^border-/, variants: ['hover', 'focus'], }, ]
```

**Reply by Shadecute** [reply]
Thank you very much, my problem was solved with your help.

**Reply by leojpod** [reply]
you're welcome :)
